os/arch/arm/src/amebad/amebad_spi.c

```diff
@@ -1092,7 +1120,24 @@ static void amebad_spi_bus_initialize(struct amebad_spidev_s *priv)
 	spi_init(&priv->spi_object, priv->spi_mosi, priv->spi_miso, priv->spi_sclk, priv->spi_cs);
 
 	spi_format(&priv->spi_object, priv->nbits, priv->mode, priv->role);
+	/* Initialize the SPI semaphore that enforces mutually exclusive access */
+
+    sem_init(&priv->exclsem, 0, 1);
+
+#ifdef CONFIG_SPI_CS
 
+	//PA_15 as SPI1_CS_0
+	gpio_init(&priv->gpio_cs0, PA_15);
+	gpio_write(&priv->gpio_cs0, 1);
+	gpio_dir(&priv->gpio_cs0, PIN_OUTPUT);
+	gpio_mode(&priv->gpio_cs0, PullNone);
+
+	//PB_3 as SPI1_CS_1
+	gpio_init(&priv->gpio_cs1, PB_3);
+	gpio_write(&priv->gpio_cs1, 1);
+	gpio_dir(&priv->gpio_cs1, PIN_OUTPUT);
+	gpio_mode(&priv->gpio_cs1, PullNone);
+#endif
```

Comment:
In the else case please do the single slave gpio init as I have mentioned above.